ui/src/definitely-typed/filters/definitelyTypedFilters.tsx

```diff
@@ -2,80 +2,174 @@ import React, { useEffect, useState } from 'react';
+      isShowingWarnings: isShowing === 'warning',
+      isShowingSuccesses: isShowing === 'success',
     };
+  }
+
+  function buildFilterInOption() {
+    if (isShowing === 'error') {
+      return errorPresetFilterId === 1 && filterIn
+        ? new RegExp(filterIn, 'i')
+        : errorPresetFilterId === 0
+        ? null
+        : errorPresetFilterId !== 1
+        ? errorPresetFilters[errorPresetFilterId - 3].regex
+        : null;
+    }
+
+    return filterIn ? new RegExp(filterIn, 'i') : null;
+  }
 
-    const infoPanel: JSX.Element[] = infoKeys.map((infoKey: keyof DefinitelyTypedRunInfo) => {
-        const selectedClass = infoKey === isShowing ? ' selected' : '';
+  const infoKeys: Array<keyof DefinitelyTypedRunInfo> = [
+    'total',
+    'success',
+    'warning',
+    'error',
+  ];
 
-        return <div key={infoKey} className={'DefinitelyTypedFilters-' + infoKey + selectedClass} onClick={() => setIsShowing(infoKey)}>
-            <span className='DefinitelyTypedFilters-infoLabel'>{infoLabelMap[infoKey]}</span>
-            <span className='DefinitelyTypedFilters-infoCount'>{props.runInfo[infoKey]}</span>
+  const infoLabelMap: { [key in keyof DefinitelyTypedRunInfo]: string } = {
+    error: 'Errors',
+    success: 'Passed',
+    warning: 'Warnings',
+    total: 'Total Types',
+  };
+
+  const infoPanel: JSX.Element[] = infoKeys.map(
+    (infoKey: keyof DefinitelyTypedRunInfo) => {
+      const selectedClass = infoKey === isShowing ? ' selected' : '';
+
+      return (
+        <div
+          key={infoKey}
+          className={'DefinitelyTypedFilters-' + infoKey + selectedClass}
+          onClick={() => setIsShowing(infoKey)}
+        >
+          <span className="DefinitelyTypedFilters-infoLabel">
+            {infoLabelMap[infoKey]}
+          </span>
+          <span className="DefinitelyTypedFilters-infoCount">
+            {props.runInfo[infoKey]}
+          </span>
         </div>
-    });
-
-    return <div className='DefinitelyTypedFilters-container'>
-        <div className='DefinitelyTypedFilters-controlsContainer'>
-            <div className='DefinitelyTypedFilters-inputsContainer'>
-                <div className='DefinitelyTypedFilters-input'>
-                    <p>Filter out messages (regex)</p>
-                    <input className='Input' type='text' value={filterOut} onChange={e => setFilterOut(e.target.value)} />
-                </div>
-                <div className='DefinitelyTypedFilters-input'>
-                    <p>Filter in only messages (regex)</p>
-                    <input className='Input' type='text' value={filterIn} onChange={e => setFilterIn(e.target.value)} />
-                </div>
+      );
+    }
+  );
+
+  type PresetFilterOption = PresetFilter & { disabled?: never; };
+
+  let errorPresetFilters: PresetFilterOption[] = [];
+
+  if (isShowing === 'error') {
+    errorPresetFilters = errorPresetFilters.filter(filter => props.data.some(d => filter.regex.test(d.message || '')));
```

Comment:
I think you wanted to use ErrorPresetFilters instead of errorPresetFilters

```
errorPresetFilters = ErrorPresetFilters.filter(filter => 
```

Reply:
done, I'm not sure what happened here...